### PR TITLE
TFSID 888124 :[SDK] Update custom start of week date functions samples

### PR DESCRIPTION
**Pull request by saugatt**
Remove all except the custom start of the week for date functions with `localstr` as the last argument. See https://github.com/tableau/connector-plugin-sdk/issues/505 for details

### CODE REVIEW

**Comment by rosswbrown** [comment]
In Annotated.tdd update the formula's for the custom SOW ones to use <formula part='week'> instead of <formula>, [example](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/Annotated.tdd#L996).  It would work as is today I think, but this will make it more precise.

File: samples/components/dialects/Annotated.tdd [hunk 1]
```diff
@@ -1006,7 +1008,8 @@
       <argument type='datetime' />
     </date-function>
     <date-function name='DATEPART' return-type='int'>
-      <formula>FLOOR((7 + TO_NUMBER(TO_CHAR(%2,'DDD')) - 1 + MOD(7 + MOD(TO_NUMBER(TO_CHAR(TRUNC(CAST(%2 AS DATE), 'YYYY'), 'J')) + 1, 7) - %3, 7)) / 7)</formula>
+      <formula>TO_NUMBER(TO_CHAR(%2,&apos;%1&apos; %3&apos;))</formula>
```
Comment (rosswbrown) [inline samples/components/dialects/Annotated.tdd:1011]: remove `<formula>TO_NUMBER(TO_CHAR(%2,&apos;%1&apos; %3&apos;))</formula>`, only need `<formula part='week'> ...` for custom SOW paths.

Same comment for others in this file and full_dialect.tdd

File: samples/components/dialects/full_dialect.tdd [hunk 2]
```diff
@@ -1629,67 +1629,31 @@
     </date-function>
     <date-function name='DATEDIFF' return-type='int'>
       <formula>DATEDIFF(%1,%2,%3)</formula>
```
Comment (rosswbrown) [inline samples/components/dialects/full_dialect.tdd:1631]: Needs same change as Annotated.tdd

Comment (saugatt) [reply samples/components/dialects/full_dialect.tdd]: Added the changes

**Pull request merged**